control_toolbox/include/control_toolbox/pid.hpp

```diff
@@ -125,53 +144,131 @@ class Pid
+      saturation_(false),
+      antiwindup_(antiwindup),
+      antiwindup_strat_("none")
+    {
+    }
+
+    /*!
+   * \brief Optional constructor for passing in values
+   *
+   * \param p The proportional gain.
+   * \param i The integral gain.
+   * \param d The derivative gain.
+   * \param i_max Upper integral clamp.
+   * \param i_min Lower integral clamp.
+   * \param u_max Upper output clamp.
+   * \param u_min Lower output clamp.
+   * \param trk_tc Specifies the tracking time constant for the 'back_calculation'
+        and 'conditioning_technique' strategies. If set to 0.0 when one of these
+        strategies is selected, a recommended default value will be applied.
+   * \param saturation Enables output saturation. When true, the controller output is
+        clamped between u_max and u_min.
+   * \param antiwindup Anti-windup functionality. When set to true, limits
+        the integral error to prevent windup; otherwise, constrains the
+        integral contribution to the control output. i_max and
+        i_min are applied in both scenarios.
+   * \param antiwindup_strat Choose the anti-windup strategy. Options: 'back_calculation',
+        'conditioning_technique', 'conditional_integration', or 'none'. Note that
+        the 'back_calculation' and 'conditioning_technique' strategies use the
+        tracking_time_constant parameter to tune the anti-windup behavior. When a strategy
+        other than 'none' is selected, it will override the controller's default anti-windup behavior.
+   *
+   */
+    Gains(
+      double p, double i, double d, double i_max, double i_min, double u_max, double u_min,
+      double trk_tc, bool saturation, bool antiwindup, std::string antiwindup_strat)
+    : p_gain_(p),
+      i_gain_(i),
+      d_gain_(d),
+      i_max_(i_max),
+      i_min_(i_min),
+      u_max_(u_max),
+      u_min_(u_min),
+      trk_tc_(trk_tc),
+      saturation_(saturation),
+      antiwindup_(antiwindup),
+      antiwindup_strat_(antiwindup_strat)
     {
     }
 
     // Default constructor
-    Gains() : p_gain_(0.0), i_gain_(0.0), d_gain_(0.0), i_max_(0.0), i_min_(0.0), antiwindup_(false)
+    Gains()
+    : p_gain_(0.0),
+      i_gain_(0.0),
+      d_gain_(0.0),
+      i_max_(0.0),
+      i_min_(0.0),
+      u_max_(0.0),
+      u_min_(0.0),
+      trk_tc_(0.0),
+      saturation_(false),
+      antiwindup_(false),
+      antiwindup_strat_("none")
     {
     }
 
-    double p_gain_;   /**< Proportional gain. */
-    double i_gain_;   /**< Integral gain. */
-    double d_gain_;   /**< Derivative gain. */
-    double i_max_;    /**< Maximum allowable integral term. */
-    double i_min_;    /**< Minimum allowable integral term. */
-    bool antiwindup_; /**< Antiwindup. */
+    double p_gain_;
+    double i_gain_;
+    double d_gain_;
+    double i_max_;
+    double i_min_;
+    double u_max_;
+    double u_min_;
+    double trk_tc_;
+    bool saturation_;
+    bool antiwindup_;
+    std::string antiwindup_strat_;
```

Comment:
I think it make sense to re-add the code comments deleted in https://github.com/ros-controls/control_toolbox/commit/f561b704ceb9bbb54a0a68edca5ccc12a953b71e, what do you think?